src/plugins/auto_batch/src/plugin.cpp

```diff
@@ -7,353 +7,395 @@
                                            batched_inputs,
-                                           batched_outputs);
-}
-
-InferenceEngine::IExecutableNetworkInternal::Ptr Plugin::LoadExeNetworkImpl(
-    const InferenceEngine::CNNNetwork& network,
-    const std::shared_ptr<InferenceEngine::RemoteContext>& context,
-    const std::map<std::string, std::string>& user_config) {
-    return LoadNetworkImpl(network, context, user_config);
+                                           batched_outputs,
+                                           compiled_model_with_batch,
+                                           compiled_model_without_batch,
+                                           context);
 }
 
-InferenceEngine::QueryNetworkResult Plugin::QueryNetwork(const InferenceEngine::CNNNetwork& network,
-                                                         const std::map<std::string, std::string>& user_config) const {
-    auto core = GetCore();
-    if (!core)
-        return InferenceEngine::QueryNetworkResult();
-    auto cfg = user_config;
+ov::SupportedOpsMap Plugin::query_model(const std::shared_ptr<const ov::Model>& model,
+                                        const ov::AnyMap& properties) const {
+    OPENVINO_ASSERT(model, "OpenVINO Model is empty!");
+    OPENVINO_ASSERT(get_core(), "Core is missing!");
+    auto cfg = properties;
     for (const auto& c : cfg) {
+        OPENVINO_SUPPRESS_DEPRECATED_START
         if (c.first == CONFIG_KEY(AUTO_BATCH_DEVICE_CONFIG) || c.first == ov::device::priorities.name()) {
             auto val = c.second;
             cfg.erase(c.first);
-            auto metaDevice = ParseMetaDevice(val, cfg);
-            return core->QueryNetwork(network, metaDevice.device_name, cfg);
+            auto metaDevice = parse_meta_device(val.as<std::string>(), cfg);
+            return get_core()->query_model(model, metaDevice.device_name, cfg);
         }
+        OPENVINO_SUPPRESS_DEPRECATED_END
     }
-    IE_THROW() << "Value for KEY_AUTO_BATCH_DEVICE_CONFIG is not set";
+    OPENVINO_THROW("Value for ov::device::priorities for AUTO BATCH PLUGIN is not set");
+}
+
+std::shared_ptr<ov::IRemoteContext> Plugin::get_default_context(const ov::AnyMap& remote_properties) const {
+    auto full_properties = merge_properties(m_plugin_config, remote_properties);
+    OPENVINO_SUPPRESS_DEPRECATED_START
+    auto it = full_properties.find(CONFIG_KEY(AUTO_BATCH_DEVICE_CONFIG));
+    OPENVINO_SUPPRESS_DEPRECATED_END
+    if (it == full_properties.end())
+        it = full_properties.find(ov::device::priorities.name());
+    if (it == full_properties.end())
+        OPENVINO_THROW("Value for ov::device::priorities is not set");
+
+    auto val = it->second.as<std::string>();
+    auto metaDevice = parse_meta_device(val, ov::AnyMap());
+    full_properties.erase(it);
+    auto remote_context =
+        std::make_shared<ov::autobatch_plugin::RemoteContext>(get_core()->get_default_context(metaDevice.device_name));
```

Comment:
@ilyachur 
When auto batch plugin call `get_default_context()`, auto batch plugin will through call the core level 'get_default_context()' with device name to get the result. But core level 'get_default_context()' return value type is '**ov::RemoteContext**' not `ov::**I**RemoteContext`, which is the `impl` in '**ov::RemoteContext**'.
And the plugin level `get_default_context()` return type is `ov::**I**RemoteContext`, so I have to use friendly class to access the 'impl' in '**ov::RemoteContext**'.
As you said, I wrap the `ov::**I**RemoteContext` , but this can not avoid to access the `impl`. 
Maybe I lost some points. Need your help here. SOS!!!

Reply:
You don't need to have a access to low level plugin remote context implementation if you provide auto plugin implementation.
Just wrap ov::RemoteContext inside AutoBatchRemoteContext and recall methods of plugin context inside the implementation of auto batch remote context.

Reply:
Haha....mislead by inherit from `ov::IRemoteContext`. If by `ov::RemoteContext`, it will be easy. Thanks Again!